src/components/taglib/TransformHelper/handleComponentKeyAttrs.js

```diff
@@ -0,0 +1,43 @@
+"use strict";
+
+const keyPostfix = /:key$/;
+
+module.exports = function handleComponentKeyAttrs() {
+    var el = this.el;
+    var context = this.context;
+
+    // BEGIN support for deprecated for attributes
+
+    let deprecatedForAttributes = ['for-ref', 'for-key', 'w-for'];
+
+    deprecatedForAttributes.forEach(attributeName => {
+        if (el.hasAttribute(attributeName)) {
+            context.deprecate(`The "${attributeName}" tag is deprecated. Please use "for:key" instead.`);
+
+            let incompatibleAttributes = ['for', 'for:key'].concat(deprecatedForAttributes.filter(a => a != attributeName)).filter(a => el.hasAttribute(a));
+            if (incompatibleAttributes.length) {
+                this.addError(`The "${attributeName}" tag cannot be used with "${incompatibleAttributes.join('" or "')}".`);
+            } else {
+                el.setAttributeValue('for:key', el.getAttributeValue(attributeName));
+            }
+
+            el.removeAttribute(attributeName);
+        }
+    });
+
+    // END support for deprecated for attributes
+
+    el.attributes.forEach(attribute => {
+        if (keyPostfix.test(attribute.name)) {
```

Comment:
Using `.endsWith(...)` is much faster than testing the regex. I would recommend:

```js
const keySuffix = ':key';

module.exports = function handleComponentKeyAttrs() {
    ...
    el.attributes.forEach(attribute => {
        const attributeName = attributeName;
        if (attributeName !== keySuffix && attributeName.endsWith(keySuffix)) {
            // Make your dreams come true here...
        }
    });
}